File: competence-be/src/main/java/com/project/competence/employee/dto/CreateEmployeeRequest.java
package com.project.competence.employee.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Builder;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

@Builder
public record CreateEmployeeRequest(
        @NotBlank String name,
        @NotBlank String surname,
        @NotNull LocalDate dateOfEmployment,
        UUID managerId,
        @NotNull List<SkillResource> skills,
        @NotNull List<ProjectResource> projects
) {
}

File: competence-fe/src/app/features/employee/dto/create-employee-request.ts
import { EmployeeModel } from '../models/employee.model';
import { ProjectModel } from '../models/project.model';

export interface CreateEmployeeRequest {
  name: string;
  surname: string;
  dateOfEmployment: Date;
  manager: EmployeeModel | null;
  skills: string[];
  projects: ProjectModel[];
}
